fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/domain/JobExecutionRepository.java

```diff
@@ -165,30 +170,85 @@ public void updateJobStatusToFailed(Long stuckJobId, String partitionerStepName)
+                        """
+                                SELECT PARAMS.parameter_value FROM json_table(
+                                (SELECT
+                                    CJP.parameter_json
+                                FROM
+                                    BATCH_JOB_INSTANCE BJI
+                                        INNER JOIN BATCH_JOB_EXECUTION BJE ON BJI.JOB_INSTANCE_ID = BJE.JOB_INSTANCE_ID
+                                        INNER JOIN BATCH_JOB_EXECUTION_PARAMS BJEP ON BJE.JOB_EXECUTION_ID = BJEP.JOB_EXECUTION_ID
+                                        inner join batch_custom_job_parameters CJP ON BJEP.parameter_value = CJP.id
+                                        AND BJEP.parameter_name = :jobCustomParamKeyName
+                                WHERE
+                                    json_contains(CJP.parameter_json, :targetJsonValue, '$')
+                                  AND BJE.STATUS IN (:statuses)
+                                  AND BJI.JOB_NAME = :jobName
+                                  AND BJE.JOB_INSTANCE_ID NOT IN (
+                                    SELECT
+                                    IBJE.JOB_INSTANCE_ID
+                                    FROM
+                                    BATCH_JOB_INSTANCE IBJI
+                                    INNER JOIN BATCH_JOB_EXECUTION IBJE ON IBJI.JOB_INSTANCE_ID = IBJE.JOB_INSTANCE_ID
+                                    WHERE IBJE.STATUS IN (:completedStatuses)
+                                  AND IBJI.JOB_NAME = :jobName
+                                    )),
+                                    "$[*]" COLUMNS(parameter_name VARCHAR(100) PATH "$.parameterName", parameter_value VARCHAR(100) PATH "$.parameterValue"))
+                                    AS PARAMS
+                                WHERE PARAMS.parameter_name = 'BusinessDate';
+                                """,
+                        Map.of("jobCustomParamKeyName", jobCustomParamKeyName, "targetJsonValue",
+                                "{\"parameterName\":\"" + parameterKeyName + "\",\"parameterValue\":\"" + parameterValue + "\"}",
+                                "statuses", List.of(STARTED.name(), STARTING.name()), "completedStatuses",
+                                List.of(COMPLETED.name(), FAILED.name(), UNKNOWN.name()), "jobName", jobName),
+                        String.class);
+            } else {
+                throw new IllegalStateException("Database type is not supported for json query " + databaseTypeResolver.databaseType());
+            }
+        } catch (EmptyResultDataAccessException e) {
+            return null;
         }
+        if (StringUtils.isBlank(businessDateInString)) {
+            throw BusinessDateNotFoundException.notFound(BusinessDateType.BUSINESS_DATE.name());
+        }
+        return LocalDate.parse(businessDateInString, DateTimeFormatter.ISO_DATE);
```

Comment:
Can you please add some tests to cover the functionality?

Reply:
we already have integration test that covers the catch-up mechanism